projectblacklight/spotlight
# Enable selection of image area for non-embed item widgets (other than uploaded item)

Closes #3186 

What this pull request does:

- Sets up an image selection link for resources, with a flag indicating if a specific resource type (e.g. embed or uploaded item) should or should not display (this version of) the image selection link.  The link passes along the id of the block item as well as the index of the item (e.g. the block item corresponds to the first item row widget on the page, whereas the index corresponds to the specific item that has been selected from autocomplete)
- Sets up a modal that opens on clicking the image selection link.  
- Updates the crop module and some of the other pages that also use the crop module to enable a consistent mechanism for identifying the IIIF fields needed for the cropping functions.  Also allows for not preserving the aspect ratio of the original crop box for the item widgets (while preserving that functionality for masthead/thumbnail cropping)
- Sets up listeners for opening and saving the modal that are able to transfer information properly for the IIIF fields between the modal and the widget form.


How to test:

- Click edit on the home page. 
- Add an item row (or item grid or item carousel or item slideshow) widget.
- Select an item from the autocomplete
- Click "select image area" under the item
- Select an area in the cropping tool
- Click "save changes" on the modal

The thumbnail should change to reflect the cropping selection. 

- Click save changes on the form.

The home page should now show the image from the added item where the image should display according to the cropped selection.
